pkg/webhook/resources/virtualmachinerestore/validator.go

```diff
@@ -305,3 +306,21 @@ func (v *restoreValidator) checkVolumeSnapshotClass(vmBackup *v1beta1.VirtualMac
 	}
 	return nil
 }
+
+func (v *restoreValidator) Delete(_ *types.Request, newObj runtime.Object) error {
+	vmRestore := newObj.(*v1beta1.VirtualMachineRestore)
+	vm, err := v.vms.Get(vmRestore.Namespace, vmRestore.Spec.Target.Name)
```

Comment:
@WebberHuang1118 

A question: if one un-restored VM happens to have the same ns & name, what will happen?

are there any other evidence like label/annotation/id... for further check? thanks.

Reply:
Hi @w13915984028,

> A question: if one un-restored VM happens to have the same ns & name, what will happen?

This is not going to happen for an existing un-restored VM that has the same ns/name as the restored VM.

If the restoring process starts first, the un-restored VM creation will be ejected, and vice versa.

Reply:
@WebberHuang1118 Thanks. I have following question :)

Assume user cluster is v1.3.1:
(1) VM A was restored from VM restore.
(2) VM A was deleted later.  ( will the VM Restore CRD be deleted as wll? when true, then it is good)
(3) A new VM A was created.

Then user upgrades to v1.4.1
(1) User try to remove the VM restore CRD object now.

In short, for many long-running and upgraded cluster, do we still have enough evidence about the VM and VM Restore relationship just by ns&name?

Reply:
> @WebberHuang1118 Thanks. I have following question :)
> 
> Assume user cluster is v1.3.1: (1) VM A was restored from VM restore. (2) VM A was deleted later. ( will the VM Restore CRD be deleted as well? when true, then it is good) (3) A new VM A was created.
> 
> Then user upgrades to v1.4.1 (1) User try to remove the VM restore CRD object now.
> 
> In short, for many long-running and upgraded clusters, do we still have enough evidence about the VM and VM Restore relationship just by ns&name?

Hi @w13915984028 

During step  `(2) VM A was deleted later. ( will the VM Restore CRD be deleted as well? when true, then it is good)`, the VMRestore CR will be deleted (as your expectation), since we have owner reference on it (which points to the restored VM).